src/cache/SimpleCache.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// Immutable is an optional dependency
+// eslint-disable-next-line import/no-extraneous-dependencies
+import {
+  IS_LOADING,
+  SET_DATA,
+  DELETE_DATA,
+  ERROR,
+  CLEAR_ERROR,
+} from './constants';
+
+function reducer(state = {
+  errors: {},
+  loading: {},
+  data: {},
+}, action) {
+  const [namespace] = action.type.split('/');
+  if (namespace !== '@fetchye') {
+    return state;
+  }
+  switch (action.type) {
+    case DELETE_DATA: {
+      const { [action.hash]: deletedEntry, ...nextData } = state.data;
```

Comment:
Love this fancy way of removing properties using destructuring... my fav single liner